kcqrs-core/src/main/kotlin/com/clouway/kcqrs/core/SimpleAggregateRepository.kt

```diff
@@ -49,7 +49,8 @@ class SimpleAggregateRepository(private val eventStore: EventStore,
             is SaveEventsResponse.SnapshotRequired -> {
                 val currentAggregate = buildAggregateFromHistory(aggregateClass, response.currentEvents, aggregate.getId()!!, response.currentSnapshot)
 
-                val newSnapshot = currentAggregate.getSnapshotMapper().toSnapshot(currentAggregate)
+                val newSnapshot = Snapshot(currentAggregate.getExpectedVersion(), Binary(messageFormat.formatToString(currentAggregate)))
```

Comment:
This couples the structure of the aggregate with the snapshot which doesn't sound like a good idea. getSnapshotMapper method could accept AggregateContext which to contain a reference to the MessageFormat. This will let the client code deside how and what snapshot to create.

Reply:
```
messageFormat.formatToString(currentAggregate)
```
the snapshot has a string content of what actually comes. I cannot understand your comment.
Maybe we can provide an interface SnapshotTransformer and use it instead of the messageFormat.

Reply:
The idea is that different object could be used for the snapshot instead of the aggregate itself. The aggregate structure will change with time which means that there could be many versions of the Snapshot.

// get mapper which uses transfer to map -> the library enforces specific transformer.  
getSnapshotMapper(SnapshotTransformer)

// get mapper which uses the provided snapshot context  -> context is more implicit and the client code may use it or may not use it. 
getSnapshotMapper(SnapshotContext)